[Serve] Pin Pydantic Version

```
➜  ~/Desktop/ray/ray/python/ray/serve/tests git:(master) ✗ pytest -v -s test_deploy.py -k handle
ImportError while loading conftest '/Users/simonmo/Desktop/ray/ray/python/ray/serve/tests/conftest.py'.
conftest.py:8: in <module>
    from ray import serve
../__init__.py:1: in <module>
    from ray.serve.api import (
../api.py:15: in <module>
    from ray.serve.common import EndpointTag
../common.py:7: in <module>
    from ray.serve.config import BackendConfig, ReplicaConfig
../config.py:6: in <module>
    from pydantic import BaseModel, PositiveInt, validator, NonNegativeFloat
E   ImportError: cannot import name 'NonNegativeFloat'
```

--- a/python/setup.py
+++ b/python/setup.py
@@ -96,7 +96,7 @@ ray_files += [
 # in this directory
 extras = {
     "full": ["colorful"],
-    "serve": ["uvicorn", "requests", "pydantic", "starlette", "fastapi"],
+    "serve": ["uvicorn", "requests", "pydantic>=1.8", "starlette", "fastapi"],
     "tune": ["pandas", "tabulate", "tensorboardX"],
     "k8s": ["kubernetes"]
 }

